### PR TITLE
heltec esp32s3 wifi lora v3: new espressif board

### PR DESCRIPTION
**Pull request by imnotjames**
adds the [Heltec ESP32S3 WiFi LoRa V3](https://heltec.org/project/wifi-lora-32-v3/) as a supported espressif ESP32-S3FN8 board

this board features a Semtec SP1261 LoRa chip (over SPI) and an 0.96-inch 128*64 screen (over I2C)

the board doesn't connect USB by default and instead supports interactions via a CP210x over USB

the USB VID/PID was pulled from [the creator's repository](https://github.com/Heltec-Aaron-Lee/WiFi_Kit_series/blob/master/esp32/boards.txt#L28-L29) but it's the standard espressif VID and a PID I don't recognize

[Datasheet can be found here](https://resource.heltec.cn/download/WiFi_LoRa32_V3/HTIT-WB32LA_V3(Rev1.1).pdf)

### CODE REVIEW

**Comment by imnotjames** [comment]
Do I need to request a `CIRCUITPY_CREATION_ID` somewhere for this?  Otherwise Bluetooth doesn't seem to build.

**Comment by dhalbert** [comment]
> Do I need to request a `CIRCUITPY_CREATION_ID` somewhere for this? Otherwise Bluetooth doesn't seem to build if I omit it.

https://github.com/creationid/creators

**Comment by imnotjames** [comment]
> > Do I need to request a `CIRCUITPY_CREATION_ID` somewhere for this? Otherwise Bluetooth doesn't seem to build if I omit it.
> 
> https://github.com/creationid/creators

Got it.  Since this already has a USB VID and PID from espressif should I use the community documented VID & PID for espressif s3 boards?

**Comment by dhalbert** [comment]
The USB VID/PID are irrelevant since they will not be used. The creator ID should be based on the actual manufacturer, in this case, Heltec. When possible, we use the manufacturer's USB VID as part of the creator ID, but it sounds like they don't have their own VID? So just make up a creator ID -- anything that makes sense. Then for the creation ID, you could also see if there's a part number or something, or else just choose one. You might get some ideas from the other ones.

**Comment by imnotjames** [comment]
> The USB VID/PID are irrelevant since they will not be used. The creator ID should be based on the actual manufacturer, in this case, Heltec. When possible, we use the manufacturer's USB VID as part of the creator ID, but it sounds like they don't have their own VID? So just make up a creator ID -- anything that makes sense. Then for the creation ID, you could also see if there's a part number or something, or else just choose one. You might get some ideas from the other ones.

Got it!  I have updated this with the proposed creator ID & creation ID that I opened as a PR over in the creators repo

File: ports/espressif/boards/heltec_esp32s3_wifi_lora_v3/sdkconfig [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#
+# Espressif IoT Development Framework Configuration
+#
+#
+# Component config
+#
+#
+# LWIP
+#
+CONFIG_ESP_PHY_ENABLE_USB=y
```
Comment (tannewt) [inline ports/espressif/boards/heltec_esp32s3_wifi_lora_v3/sdkconfig:10]: I doubt you need this if the native USB isn't connected.

**Review by tannewt — CHANGES_REQUESTED** [review]
One sdkconfig piece I don't think you need.

File: ports/espressif/boards/heltec_esp32s3_wifi_lora_v3/sdkconfig (hunk 1 above)
Comment (imnotjames) [reply ports/espressif/boards/heltec_esp32s3_wifi_lora_v3/sdkconfig:10]: Hmm, probably not.  I thought I enabled it for a reason but I can't remember why.

I'll try disabling it, building, and flashing.

If it works fine I'll remove it.  If it doesn't, I'll add a comment describing what reason it should be enabled.

Comment (imnotjames) [reply ports/espressif/boards/heltec_esp32s3_wifi_lora_v3/sdkconfig:10]: Loads just fine, not sure where I thought I needed it.  Removing!

**Review by tannewt — APPROVED** [review]

**Pull request merged**